plugins/bookmarks/__init__.py

```diff
@@ -244,84 +230,73 @@ def delete_bookmark(self, widget, targets):
 
         self.save_db()
 
+    @common.idle_add()
```

Comment:
> Because it does I/O operations which might block, e.g. during swapping/paging or because of disk suspend.

Maybe it should be threaded instead…. Does python have async I/O? Or should I use GLib's GIO instead? I've read that the latter does most stuff async so it should be safe here.
If you are fine with the other changes, I could cherry-pick them on master and fix this one only.